src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/config/BailNotificationHandlerConfiguration.java

```diff
@@ -34,6 +36,38 @@ public PreSubmitCallbackHandler<BailCase> submitApplicationHearingCentreNotifica
         );
     }
 
+    @Bean
+    public PreSubmitCallbackHandler<BailCase> uploadSummaryNotificationHandler(
+            @Qualifier("uploadSummaryNotificationGenerator") List<BailNotificationGenerator> bailNotificationGenerators
+    ) {
+        return new BailNotificationHandler(
+                (callbackStage, callback) -> {
+                    BailCase bailCase = callback.getCaseDetails().getCaseData();
```

Comment:
Add a check in there:
 boolean isAllowedBailCase = (callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
                            && callback.getEvent() == Event.UPLOAD_BAIL_SUMMARY);
if (isAllowedBailCase) {
       BailCase bailCase = callback.getCaseDetails().getCaseData();
      return (callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
                            && callback.getEvent() == Event.UPLOAD_BAIL_SUMMARY
                            && isLegallyRepresented(bailCase));
} else 
return isAllowedBailCase;